### DISCUSSION TITLE
I dont understand the JBD Blocking Charge & AllowtoDischarge values

### DISCUSSION THREAD

**Discussion by optio50** [Louisvdw/dbus-serialbattery#217] [Q&A] [answered]
/System/NrOfModulesBlockingCharge
Io/AllowToCharge
Io/AllowToDischarge
&
Info/MaxChargeCurrent

I can charge and discharge but have a value of 1 in /System/NrOfModulesBlockingCharge
Also have conflicting values in  Io/AllowToCharge Io/AllowToDischarge

In addition,  Info/MaxChargeCurrent says 5.
I can and do charge above 5 amps.

See screenshot

![Screenshot_JBD-BMS](https://user-images.githubusercontent.com/4461936/192071250-b78a818b-c9a1-44a4-ad11-8286b5e68124.png)

**Comment by Louisvdw** [comment, marked as answer]
When your battery is at 100% it will limit your charge to 5A.
Use your battery to below 99% and your charge current limit will start to increase
https://github.com/Louisvdw/dbus-serialbattery/wiki/Features#charge-current-control-management

**Reply by optio50** [reply]
Thanks Louis

Can you also tell me why Io/AllowToCharge would be turned off?
I still have current going into the battery yet AllowToCharge is off
![Screenshot_JBD-BMS2](https://user-images.githubusercontent.com/4461936/192118285-4093c18a-0797-4ddb-b4a3-8d7f98b13914.png)

**Reply by optio50** [reply]
I think I had DVCC turned off. That may have caused it.

**Reply by Louisvdw** [reply]
The logic you will find in battery.py function manage_charge_current
https://github.com/Louisvdw/dbus-serialbattery/blob/master/etc/dbus-serialbattery/battery.py#:~:text=def%20manage_charge_current
